pkg/metrics/histogram_bucket.go

```diff
@@ -33,9 +33,9 @@ func (m *HistogramBucket) GetHost() string {
 }
 
 // GetTags returns the bucket tags.
-func (m *HistogramBucket) GetTags(tb *tagset.HashingTagsAccumulator) {
+func (m *HistogramBucket) GetTags(taggerBuffer, metricBuffer *tagset.HashingTagsAccumulator) {
```

Comment:
why pass taggerBuffer here?  It's shared, so it would be pretty bad if this function modified it, right?

Reply:
It's part of the interface. If HistogramBuckets supported origin detection, this function would be required to write tags from the origin detection to this accumulator. The contract is indeed to modify the accumulator, so it shouldn't be shared.

Reply:
If I understand correctly, the `GetTags(*tagset.HashingTagsAccumulator, *tagset.HashingTagsAccumulator)` doesn't guard against an implementation modifying the `taggerBuffer` which should not be modified, right?

Reply:
No, it is very much should to be modified. It comes in empty and the implementation should fill it with tags from origin detection. (Doesn't matter too much which one even, except we keep tags in the first one when there are duplicates).